### DISCUSSION TITLE
Can Thunder client sign soap request with a private key (.p12) just like Jmeter?

### DISCUSSION THREAD

**Discussion by aba-GitHubUser** [thunderclient/thunder-client-support#625] [Q&A]
I was wondering if it is possible to sign xml soap requests in **Thunder client** just like in other tools, **SoapUI** and **Jmeter**.
Normally, some soap header elements and the body could be signed with a private key in **Jmeter**.
If this feature is not yet implemented, is it at all possible to integrate (this feature) in the future?

Thanks.

**Comment by rangav** [comment]
@aba-GitHubUser you can use POST request and send xml body and add any additional headers needed.

if you have any public sample api, I can verify it.

**Reply by aba-GitHubUser** [reply]
Thank you for your answer, Mr Ranga Vadhineni.
Yes I know I could send soap requests with Thunder Client, as a matter of fact I did play with
this functionality. But I was wondering if I also could send signed soap messages. I think my question above was not clear enough. 
I would like to be able to sign the soap message and send it. To give you an example, there is a plugin in Jmeter that adds a digital signature to the soap message.
This plugin is: **"WS Security for SOAP**" and hereby a link to some documentation: 
https://github.com/tilln/jmeter-wssecurity

This is how this plugin is used within Jmeter:

![Soap message signer](https://user-images.githubusercontent.com/43840953/173894137-5254796f-b135-47a9-bc40-2f63a677bd9e.png)

**Reply by rangav** [reply]
Thanks got it, I will review it and add to roadmap.

**Reply by aba-GitHubUser** [reply]
Thank you very much, Mr Ranga Vadhineni, for your intention to do so. Much appreciated.
